draft-ietf-ppm-dap.md

```diff
@@ -1845,22 +1845,22 @@ about the client's measurement, violating the privacy goal of DAP. To prevent
 such replay attacks, this specification requires the aggregators to detect and
 filter out replayed reports.
 
-To detect replay attacks, each aggregator keeps track of the set of report IDs
-pertaining to reports that were previously aggregated for a given task. If the
-leader receives a report from a client whose report ID is in this set, it
-either ignores it or aborts the upload sub-protocol as described in
-{{upload-flow}}. A Helper who receives an encrypted input share whose report ID
-is in this set rejects the report as described in
+To detect replay attacks in a given time window, each aggregator is required to
```

Comment:
I think we need to be more clear about what the "given time window" is. My reading of the arguments in #362 is that we're instructing implementations to choose a window of time relative to the instant at which a report being evaluated for anti-replay, and only check for unique report IDs within that window. Is the behavior the same regardless of time interval or fixed batch tasks? Is there a risk of the helper and leader applying inconsistent anti-replay policy, either because they construct their time windows differently or because the leader might be evaluating anti-replay at upload time, well before the helper gets a chance to do so at aggregation time? Finally, if we say nothing about how to construct the time windows, then doesn't that allow a conforming implementation to not do anti-replay checks at all?

Reply:
I don't think so: the point of this PR is to allow aggregators to evaluate anti-replay on a narrower scope than the entire task. Bob's [argument](https://github.com/ietf-wg-ppm/draft-ietf-ppm-dap/issues/362#issuecomment-1301108181), I think, is that aggregators only need to look at reports that wouldn't get rejected for being too old or from too far in the future, and I think perhaps this paragraph could be written in those terms.

Reply:
The "given time window" refers to the window of time for which an aggregator chooses to check for replayed reports; any reports outside of that timestamp must be rejected.

Reply:
Perfect. Then how about we just keep "given time window" as is, say that implementations can choose this to fit their needs, and add a line at the end of the paragraph saying "Any report that is outside this time window MUST be rejected" (or whatever). Would that work?

Reply:
I've pushed @chris-wood's suggestion of simply dropping discussion about the time window at that point. To @tgeoghegan point about not needing to do anti-replay for reports too far in the past or in the future, this is already premitted in {{early-input-share-validation}}.

In fact, {{early-input-share-validation}} is really the main text that matters. I think the only normative text we want to add is that anti-replay MUST be enforced per nonce, but MAY be enforced per nonce/timestamp. What do y'all think about axing this {{anti-replay}} altogether and merging this normative text into {{early-input-share-validation}}?

Reply:
If we rename that section to "Input Share Validation" and promote the replay rejection step to the first thing in the list, I'd be OK with that.

Reply:
Going with "Report Share Validation" :)

Reply:
Done. I ended up going with "Input Share Validation", as @chris-wood suggested, as this better matches the language in that section.